libs/remote_storage/src/local_fs.rs

```diff
@@ -59,22 +57,22 @@ impl LocalFs {
 
     async fn read_storage_metadata(
         &self,
-        file_path: &Path,
+        file_path: &Utf8Path,
     ) -> anyhow::Result<Option<StorageMetadata>> {
-        let metadata_path = storage_metadata_path(file_path);
+        let metadata_path = storage_metadata_path(file_path)?;
         if metadata_path.exists() && metadata_path.is_file() {
             let metadata_string = fs::read_to_string(&metadata_path).await.with_context(|| {
                 format!(
                     "Failed to read metadata from the local storage at '{}'",
-                    metadata_path.display()
+                    metadata_path.as_std_path().display()
```

Comment:
No need to go to std pathbuf here either because Utf8Path `impl std::fmt::Display`.